explore/confounding/confounding.py

```diff
@@ -0,0 +1,193 @@
+
+# coding: utf-8
+
+# # Create a logistic regression model to several mutations from covariates
+
+# In[1]:
+
+import os
+import itertools
+import warnings
+import collections
+
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+import seaborn as sns
+from sklearn import preprocessing, grid_search
+from sklearn.linear_model import SGDClassifier
+from sklearn.cross_validation import train_test_split
+from sklearn.metrics import roc_auc_score, roc_curve
+from sklearn.pipeline import make_pipeline
+from sklearn.preprocessing import StandardScaler
+from scipy.special import logit
+
+
+# In[2]:
+
+get_ipython().magic('matplotlib inline')
+plt.style.use('seaborn-notebook')
+
+
+# ## Load Data
+
+# In[3]:
+
+path = os.path.join('..', '..', 'download', 'mutation-matrix.tsv.bz2')
+Y = pd.read_table(path, index_col=0)
+
+
+# In[4]:
+
+# Read sample information and create a covariate TSV
+url = 'https://github.com/cognoma/cancer-data/raw/54140cf6addc48260c9723213c40b628d7c861da/data/covariates.tsv'
+covariate_df = pd.read_table(url, index_col=0)
+covariate_df.head(2)
+
+
+# ## Specify the type of classifier
+
+# In[5]:
+
+param_grid = {
+    'alpha': [10 ** x for x in range(-4, 2)],
+    'l1_ratio': [0, 0.05, 0.1, 0.2, 0.5, 0.8, 0.9, 0.95, 1],
+}
+
+clf = SGDClassifier(
+    random_state=0,
+    class_weight='balanced',
+    loss='log',
+    penalty='elasticnet'
+)
+
+# joblib is used to cross-validate in parallel by setting `n_jobs=-1` in GridSearchCV
+# Supress joblib warning. See https://github.com/scikit-learn/scikit-learn/issues/6370
+warnings.filterwarnings('ignore', message='Changing the shape of non-C contiguous array')
+clf_grid = grid_search.GridSearchCV(estimator=clf, param_grid=param_grid, n_jobs=-1, scoring='roc_auc')
+pipeline = make_pipeline(
+    StandardScaler(),
+    clf_grid
+)
+
+
+# ## Specify covariates and outcomes
+
+# In[6]:
+
+def expand_grid(data_dict):
+    """Create a dataframe from every combination of given values."""
+    rows = itertools.product(*data_dict.values())
+    return pd.DataFrame.from_records(rows, columns=data_dict.keys())
+
+mutations = {
+    '7157': 'TP53', # tumor protein p53
```

Comment:
pep8 says inline comments should be separated by [two spaces](https://www.python.org/dev/peps/pep-0008/#inline-comments)